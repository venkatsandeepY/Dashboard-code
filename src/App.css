/* App-specific styles */
.app-container {
  min-height: 100vh;
  background-color: #f9fafb;
}

.main-content {
  flex: 1;
  display: flex;
  flex-direction: column;
}

.content-area {
  flex: 1;
  padding: 2rem;
}

/* Custom scrollbar styles */
::-webkit-scrollbar {
  width: 8px;
}

::-webkit-scrollbar-track {
  background: #f1f1f1;
}

::-webkit-scrollbar-thumb {
  background: #888;
  border-radius: 4px;
}

::-webkit-scrollbar-thumb:hover {
  background: #555;
}

/* Responsive utilities */
@media (max-width: 768px) {
  .content-area {
    padding: 1rem;
  }
}