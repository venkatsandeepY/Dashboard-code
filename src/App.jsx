import React, { useState } from 'react';
import { BrowserRouter as Router, Routes, Route, Navigate, useLocation } from 'react-router-dom';
import Sidebar from './Sidebar';
import Header from './components/common/Header';
import Dashboard from './Dashboard';
import Status from './Status';
import Reports from './Reports';
import Feedback from './Feedback';
import Footer from './components/common/Footer';

function AppContent() {
  const [isCollapsed, setIsCollapsed] = useState(false);
  const location = useLocation();

  const handleToggleCollapse = () => {
    setIsCollapsed(!isCollapsed);
  };

  const getActiveItem = () => {
    const path = location.pathname;
    if (path === '/status') return 'status';
    if (path === '/reports') return 'reports';
    if (path === '/dashboard') return 'dashboard';
    return 'dashboard';
  };

  return (
    <div className="app-container">
      {/* Sidebar */}
      <Sidebar 
        activeItem={getActiveItem()} 
        isCollapsed={isCollapsed}
        onToggleCollapse={handleToggleCollapse}
      />
      
      {/* Main Content */}
      <div className="main-content">
        {/* Header */}
        <Header />

        {/* Content Area */}
        <main className="content-area">
          <Routes>
            <Route path="/dashboard" element={<Dashboard />} />
            <Route path="/status" element={<Status />} />
            <Route path="/reports" element={<Reports />} />
            <Route path="*" element={<Navigate to="/dashboard" replace />} />
          </Routes>
        </main>

        {/* Footer */}
        <Footer />
      </div>
    </div>
  );
}

function App() {
  return (
    <Router>
      <AppContent />
    </Router>
  );
}

export default App;