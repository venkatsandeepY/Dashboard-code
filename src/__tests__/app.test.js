import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../App.jsx';

describe('App', () => {
  it('renders key layout elements', () => {
    render(<App />);

    // Sidebar title and Header title share text 'ESQM'; ensure at least one appears
    expect(screen.getAllByText('ESQM').length).toBeGreaterThan(0);

    // Footer content
    expect(screen.getByText(/All rights reserved/i)).toBeInTheDocument();

    // Navigation labels
    expect(screen.getByText('Dashboard')).toBeInTheDocument();
    expect(screen.getByText('Status')).toBeInTheDocument();
    expect(screen.getByText('Reports')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});


