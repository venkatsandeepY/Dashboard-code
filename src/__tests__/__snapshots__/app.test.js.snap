// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`App matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app-container"
    style="margin: 0px; padding: 0px;"
  >
    <div
      class="sidebar sidebar--expanded"
    >
      <div
        class="sidebar__header"
      >
        <div
          class="flex items-start justify-start w-full"
        >
          <div>
            <div
              class="flex items-center gap-2"
            >
              <h1
                class="sidebar__title"
              >
                ESQM
              </h1>
            </div>
            <p
              class="sidebar__subtitle"
            >
              (DLIFE)
            </p>
          </div>
        </div>
      </div>
      <nav
        class="sidebar__nav"
      >
        <ul
          class="sidebar__nav-list"
        >
          <li>
            <button
              class="nav-item nav-item--active "
              title=""
            >
              <svg
                class="nav-item__icon"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 9l9-7 9 7v11a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2z"
                />
                <polyline
                  points="9 22 9 12 15 12 15 22"
                />
              </svg>
              <span
                class="nav-item__label capitalize"
              >
                Dashboard
              </span>
            </button>
          </li>
          <li>
            <button
              class="nav-item  "
              title=""
            >
              <svg
                class="nav-item__icon"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="23 6 13.5 15.5 8.5 10.5 1 18"
                />
                <polyline
                  points="17 6 23 6 23 12"
                />
              </svg>
              <span
                class="nav-item__label capitalize"
              >
                Status
              </span>
            </button>
          </li>
          <li>
            <button
              class="nav-item  "
              title=""
            >
              <svg
                class="nav-item__icon"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8z"
                />
                <polyline
                  points="14 2 14 8 20 8"
                />
                <line
                  x1="16"
                  x2="8"
                  y1="13"
                  y2="13"
                />
                <line
                  x1="16"
                  x2="8"
                  y1="17"
                  y2="17"
                />
                <polyline
                  points="10 9 9 9 8 9"
                />
              </svg>
              <span
                class="nav-item__label capitalize"
              >
                Reports
              </span>
            </button>
          </li>
          <li>
            <button
              class="nav-item  "
              title=""
            >
              <svg
                class="nav-item__icon"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 11.5a8.38 8.38 0 0 1-.9 3.8 8.5 8.5 0 0 1-7.6 4.7 8.38 8.38 0 0 1-3.8-.9L3 21l1.9-5.7a8.38 8.38 0 0 1-.9-3.8 8.5 8.5 0 0 1 4.7-7.6 8.38 8.38 0 0 1 3.8-.9h.5a8.48 8.48 0 0 1 8 8v.5z"
                />
              </svg>
              <span
                class="nav-item__label capitalize"
              >
                Feedback
              </span>
            </button>
          </li>
        </ul>
      </nav>
    </div>
    <div
      class="main-content"
      style="margin: 0px; padding: 0px;"
    >
      <header
        class="header"
      >
        <div
          class="header__content"
        >
          <div
            class="header__left"
          >
            <button
              class="btn btn--ghost btn--icon-only text-primary hover-lift mr-md"
            >
              <svg
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="18"
                  x2="6"
                  y1="6"
                  y2="18"
                />
                <line
                  x1="6"
                  x2="18"
                  y1="6"
                  y2="18"
                />
              </svg>
            </button>
            <div>
              <h1
                class="header__title"
                style="font-size: 1.125rem;"
              >
                ESQM
              </h1>
              <p
                class="header__subtitle"
              >
                Operations Dashboard
              </p>
            </div>
          </div>
          <div
            class="header__center"
          >
            <div
              class="header__search"
            >
              <div
                class="header__search-icon"
              >
                <svg
                  fill="none"
                  height="16"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <line
                    x1="21"
                    x2="16.65"
                    y1="21"
                    y2="16.65"
                  />
                </svg>
              </div>
              <input
                class="header__search-input"
                placeholder="Search..."
                type="text"
              />
            </div>
          </div>
          <div
            class="header__right"
          >
            <img
              alt="Discover Logo"
              src="/image copy.png"
              style="height: 40px; border-radius: 8px; transition: opacity 0.2s ease;"
            />
          </div>
        </div>
      </header>
      <main
        class="content-area"
      >
        <div
          class="flex items-center justify-center min-h-[60vh]"
        >
          <div
            class="text-center"
          >
            <div
              class="flex justify-center mb-6"
            >
              <div
                class="relative"
              >
                <svg
                  class="w-16 h-16 text-blue-500"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polygon
                    points="13 2 3 14 12 14 11 22 21 10 12 10 13 2"
                  />
                </svg>
                <svg
                  class="w-8 h-8 text-purple-500 absolute -bottom-2 -right-2"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <polyline
                    points="16 18 22 12 16 6"
                  />
                  <polyline
                    points="8 6 2 12 8 18"
                  />
                </svg>
              </div>
            </div>
            <h2
              class="text-4xl font-bold text-gray-900 mb-4 tracking-tight"
            >
              Development Starts Here
            </h2>
            <p
              class="text-gray-600 text-lg font-medium"
            >
              Dashboard component ready for implementation
            </p>
          </div>
        </div>
      </main>
      <footer
        class="footer"
      >
        <div
          class="footer__content"
        >
          <div
            class="footer__grid md:flex md:justify-between md:items-center"
          >
            <div
              class="footer__copyright"
            >
              © 2024 ESQM (DLIFE) • All rights reserved
            </div>
            <div
              class="footer__links"
            >
              <a
                class="footer__link"
                href="#"
              >
                Privacy Policy
              </a>
              <span
                class="footer__separator"
              >
                •
              </span>
              <a
                class="footer__link"
                href="#"
              >
                Terms of Service
              </a>
              <span
                class="footer__separator"
              >
                •
              </span>
              <a
                class="footer__link"
                href="#"
              >
                Contact
              </a>
            </div>
          </div>
        </div>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
